src/CG/CG_elem.cpp

```diff
@@ -2577,6 +2577,217 @@ bool intersectPointBox( array3D const &P, array3D const &B1, array3D const &B2,
     return true;
 }
 
+/*!
+ * Given a direction and a flag for the fornt or rear face, it returns the local index of the
+ * spacified box face.
+ *
+ * \param[in] direction is the direction normal to the face (0, 1 or 2)
+ * \param[in] front true if the face is in the front of the box
+ * \return the index of the box interface
+ */
+int getBoxFaceLocalIndex(int direction, bool front) {
+    if (direction == 0) {
+        if (front) {
+            return 1;
+        } else {
+            return 0;
+        }
+    } else if (direction == 1) {
+        if (front) {
+            return 3;
+        } else {
+            return 2;
+        }
+    } else if (direction == 2) {
+        if (front) {
+            return 5;
+        } else {
+            return 4;
+        }
+    } else {
+        assert(0);
+    }
+    return -1;
+}
+
+/*!
+ * Checks if plane intersects a box face and computes the nodes comprising the intersected
+ * face belonging to the subspace opposite of the one pointed by the plane's normal
+ *
+ * In the degenerate case where the box face belongs to the given plane the interface
+ * is not considered as intersected.
+ *
```

Comment:
This seems inconsistent with all other functions in this module.

Reply:
I kept this change. All the functions I added for intersecting pixels, triangles, and polygons are compatible with each other.